analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/InitializationVectorShouldBeRandom.cs

```diff
@@ -0,0 +1,206 @@
+                        if (symbolicValue != null && HasNotInitializedIVConstraint(symbolicValue, programState))
+                        {
+                            context.AddLocation(invocation.GetLocation());
+                        }
+                    }
+                    else if (IsInvalidIV(invocation.ArgumentList.Arguments[1], programState))
+                    {
+                        context.AddLocation(invocation.GetLocation());
+                    }
+                }
+
+                return programState;
+            }
+
+            private ProgramState ArrayCreationPostProcess(ArrayCreationExpressionSyntax arrayCreation, ProgramState programState) =>
+                semanticModel.GetTypeInfo(arrayCreation).Type.Is(KnownType.System_Byte_Array) && programState.HasValue
+                    ? programState.SetConstraint(programState.PeekValue(), ByteArraySymbolicValueConstraint.Constant)
+                    : programState;
+
+            private bool IsInvalidIV(ArgumentSyntax argument, ProgramState programState) =>
+                argument.Expression switch
+                {
+                    MemberAccessExpressionSyntax memberAccess => memberAccess.NameIs("IV")
+                                                                 && programState.GetSymbolValue(semanticModel.GetSymbolInfo(memberAccess.Expression).Symbol) is {} symbolicValue
+                                                                 && HasNotInitializedIVConstraint(symbolicValue, programState),
+                    IdentifierNameSyntax identifier => programState.GetSymbolValue(semanticModel.GetSymbolInfo(identifier).Symbol) is {} symbolicValue
+                                                       && programState.HasConstraint(symbolicValue, ByteArraySymbolicValueConstraint.Constant),
+                    _ => false
+                };
+
+            private SymbolicValue GetSymbolicValue(InvocationExpressionSyntax invocation, ProgramState programState)
+            {
+                var invocationTarget = ((MemberAccessExpressionSyntax)invocation.Expression).Expression;
+                var targetSymbol = semanticModel.GetSymbolInfo(invocationTarget).Symbol;
+
+                return programState.GetSymbolValue(targetSymbol);
+            }
+
+            private bool IsIVMemberAccess(MemberAccessExpressionSyntax memberAccess) =>
+                memberAccess.NameIs("IV")
+                && semanticModel.GetSymbolInfo(memberAccess).Symbol is {} symbol
+                && symbol.ContainingType.DerivesFrom(KnownType.System_Security_Cryptography_SymmetricAlgorithm);
+
+            private static bool HasNotInitializedIVConstraint(SymbolicValue value, ProgramState programState) =>
+                programState.Constraints[value].HasConstraint(CryptographyIVSymbolicValueConstraint.NotInitialized);
+
+            private static bool IsCreateMethod(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
```

Comment:
This is used only once, I'd embed it in the usage. As the `IsmemberAccessOnKnownType` + `Create` + `SymmetricAlgorihm` provides clear information about what's exactly going on.

For example contrary to `HasNotInitializedIVConstraint` where the name brings additional value.